fuel/datasets/hdf5.py

```diff
@@ -542,8 +542,11 @@ def get_data(self, state=None, request=None):
             data, shapes = self._out_of_memory_get_data(state, request)
         for i in range(len(data)):
```

Comment:
Because we are doing inplace modification, which makes sense in this performance-critical place.